core/src/main/java/discord4j/core/ClientBuilder.java

```diff
@@ -60,16 +63,23 @@
     private String token;
     private int shardIndex = 0;
     private int shardCount = 1;
-    private StoreService storeService;
+    private StoreServiceLoader storeServiceLoader;
+    @Nullable
+    private StoreService storeService = null;
     private FluxProcessor<Event, Event> eventProcessor;
     private Scheduler eventScheduler;
     private Presence initialPresence;
+    @Nullable
     private IdentifyOptions identifyOptions;
     private RetryOptions retryOptions;
 
     public ClientBuilder(final String token) {
         this.token = Objects.requireNonNull(token);
-        storeService = new StoreServiceLoader().getStoreService();
+        //Increase JDK store priority by default
+        Map<Class<? extends StoreService>, Integer> priority = new HashMap<>();
+        // We want almost minimum priority, so that jdk can beat no-op but most implementations will beat jdk
+        priority.put(JdkStoreService.class, (int) (Integer.MAX_VALUE - 1));
```

Comment:
This seems confusing to think a higher number = lower priority.

Reply:
This is how the OS works to schedule processes on the CPU, I think it makes the most sense.

Reply:
Big difference between low-level systems programming and high-level API abstractions. To me it doesn't make sense, this way. If I see Integer.MIN_VALUE my first instinct is not *maximum* priority.

Reply:
I recommend you use the "order" wording instead of priority to avoid these confusions.

Reply:
Originally that was the case @danthonywalker but I changed it as per @quanticc's recommendation. So i could either reword it, or restore the original functionality. Which would be preferred?

Reply:
Reword

Reply:
I don't really like the "order" wording. It sounds kinda weird to me in the context of this.

Reply:
Regardless if you prefer to keep priority or change to order, I recommend adding some constants with "highest" / "lowest" values to clarify.

Reply:
I personally think priority fits the context a little better, but I think semantics are easy to signify in the javadocs (which I hope implementors are paying attention to). I'm not necessarily against adding those magic constants, but what use would they serve, really?

Reply:
Adding the constants would avoided this very thread.